scripts/generate-bindings.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+set -e
+
+SCRIPTS_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+MONOMER_DIR=$(cd "$SCRIPTS_DIR/.." && pwd)
+CONTRACTS_DIR=$(cd "$MONOMER_DIR/contracts" && pwd)
+BINDINGS_DIR=$(cd "$MONOMER_DIR/bindings" && pwd)
+
+# Create temporary forge artifacts and cache directories
+mkdir -p "$BINDINGS_DIR/artifacts"
+mkdir -p "$BINDINGS_DIR/cache"
+FORGE_ARTIFACTS_DIR=$(cd "$BINDINGS_DIR/artifacts" && pwd)
+FORGE_CACHE_DIR=$(cd "$BINDINGS_DIR/cache" && pwd)
+
+# Check for jq, abigen, and forge installations
+if ! command -v jq &> /dev/null
+then
+    echo "jq could not be found, please install jq for your system at https://jqlang.github.io/jq/download/"
+    exit 1
+fi
+if ! command -v abigen &> /dev/null
+then
+    echo "abigen could not be found, please run 'make install-abigen'"
+    exit 1
+fi
+if ! command -v forge &> /dev/null
+then
+    echo "forge could not be found, please install 'make install-foundry'"
+    exit 1
+fi
```

Comment:
What sort of wonky failure modes might someone encounter if the wrong versions of these tools are present?

Is it worth the effort to be more specific on the checks?

Reply:
I investigated this for a little bit and think asserting version would be good to have here, however since there is a good amount of other PRs left to do for the withdrawal implementation I'm going to defer the rest of this investigation until after the rest of the implementation is done.